pkg/api/norman/customization/cluster/validator.go

```diff
@@ -746,6 +748,29 @@ func (v *Validator) validateGKEConfig(request *types.APIContext, cluster map[str
 	return nil
 }
 
+// validateGKENetworkPolicy performs validation around setting enableNetworkPolicy on GKE clusters which turns on Project Network Isolation
+func (v *Validator) validateGKENetworkPolicy(cluster map[string]interface{}, clusterSpec *v32.ClusterSpec, prevCluster *v3.Cluster) error {
+	// determine if network policy is enabled on the GKE cluster by checking the cluster spec and then the upstream spec if the field is nil (unmanaged)
+	var netPolEnabled bool
+	if clusterSpec.GKEConfig != nil && clusterSpec.GKEConfig.NetworkPolicyEnabled != nil {
+		netPolEnabled = *clusterSpec.GKEConfig.NetworkPolicyEnabled
+	} else if prevCluster != nil && prevCluster.Status.GKEStatus.UpstreamSpec != nil && prevCluster.Status.GKEStatus.UpstreamSpec.NetworkPolicyEnabled != nil {
+		netPolEnabled = *prevCluster.Status.GKEStatus.UpstreamSpec.NetworkPolicyEnabled
+	} else {
+		return nil
+	}
+
+	// network policy enabled on the GKE cluster is a prerequisite for PNI
+	if enableNetPol := cluster["enableNetworkPolicy"]; enableNetPol == true && !netPolEnabled {
+		return httperror.NewAPIError(
+			httperror.InvalidBodyContent,
+			"Network Policy support must be enabled on GKE cluster in order to enable Project Network Isolation",
```

Comment:
If the user is not using PNI, is this check still important? Does `clusterSpec.EnableNetworkPolicy` equate to PNI?

Reply:
If the user isn't using PNI, this check is not important. `clusterSpec.EnableNetworkPolicy=true` equates to PNI enabled and vice versa